### PR TITLE
Delete WebAssembly/Caching_modules

### LINKED ISSUE #14520: Content bug: Remove references to caching WebAssembly Modules in indexedDB
## What page(s) did you find the problem on?

https://developer.mozilla.org/en-US/docs/WebAssembly/Caching_modules

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/instantiate

> shared via [postMessage()](https://developer.mozilla.org/en-US/docs/Web/API/Worker/postMessage) or [cached in IndexedDB](https://developer.mozilla.org/en-US/docs/WebAssembly/Caching_modules).

There are probably more references, probably best to search for indexedDB under WebAssembly/

## What is the problem?

IndexedDB caching was removed from Firefox 4 years ago, I don't think we need to keep the documentation for it. https://bugzilla.mozilla.org/show_bug.cgi?id=1469395

### PR DESCRIPTION
**Pull request by dawei-wang**
Fixes mdn/content#14520

### CODE REVIEW

**Comment by sideshowbarker** [comment]
Given @https://github.com/mdn/content/issues/14520#issuecomment-1306404876, requested review from @hamishwillee

**Comment by hamishwillee** [comment]
@dawei-wang Thank you for this. I just put a [comment in the corresponding issue](https://github.com/mdn/content/issues/14520#issuecomment-1306404876) that I "think" that it is good to remove this material, but that I think that it is very likely that developers will still want to cache WASM modules.

So what I think we need ideally is to replace this content showing how to use some other cache than indexDB. Do you think that this is something  you could do? If would be better to do now because the existing doc is here so you could copy the current structue.

I appreciate that's a much bigger job than deleting this! 

If you don't want to do that then let me know and I'll review this on the assumption that we are deleting this material but keeping a red-link to the old doc.

**Comment by dawei-wang** [comment]
@hamishwillee I'm not familiar with caching mechanisms enough to document a new one. Let's keep a red link to the old doc.

File: files/en-us/webassembly/javascript_interface/instantiate/index.md [hunk 1]
```diff
@@ -57,11 +57,8 @@ WebAssembly.instantiate(bufferSource, importObject);
 A `Promise` that resolves to a `ResultObject` which contains two
 fields:
 
-- `module`: A [`WebAssembly.Module`](/en-US/docs/WebAssembly/JavaScript_interface/Module) object representing the
-  compiled WebAssembly module. This `Module` can be instantiated again,
-  shared via {{domxref("Worker.postMessage", "postMessage()")}} or [cached in IndexedDB](/en-US/docs/WebAssembly/Caching_modules).
-- `instance`: A [`WebAssembly.Instance`](/en-US/docs/WebAssembly/JavaScript_interface/Instance) object that contains all
-  the [Exported WebAssembly functions](/en-US/docs/WebAssembly/Exported_functions).
+- `module`: A [`WebAssembly.Module`](/en-US/docs/WebAssembly/JavaScript_interface/Module) object representing the compiled WebAssembly module. This `Module` can be instantiated again, or shared via {{domxref("Worker.postMessage", "postMessage()")}}.
```
Comment (hamishwillee) [inline files/en-us/webassembly/javascript_interface/instantiate/index.md:60]: Adding back redlink
```suggestion
- `module`: A [`WebAssembly.Module`](/en-US/docs/WebAssembly/JavaScript_interface/Module) object representing the compiled WebAssembly module. This `Module` can be instantiated again, shared via {{domxref("Worker.postMessage", "postMessage()")}}, or [cached](/en-US/docs/WebAssembly/Caching_modules).
```

**Review by hamishwillee — APPROVED** [review]
Thanks very much @dawei-wang ! I've added back a link to the now-removed doc. The rest looks great.

**Comment by hamishwillee** [comment]
OK, Merging. Thanks @dawei-wang

**Pull request merged**